application/backend/app/services/active_model_service.py

```diff
@@ -79,13 +79,25 @@ def get_loaded_inference_model(self, force_reload: bool = False) -> LoadedModel
         active_model_id = self._model_activation_state.active_model_id
         if self._loaded_model is None or self._loaded_model.id != active_model_id:
             logger.info("Loading model with ID '%s'", active_model_id)
-            model_path = self._get_model_xml_path(project_id, active_model_id)
-            self._loaded_model = LoadedModel(
-                id=self._model_activation_state.active_model_id,
-                model=Model.create_model(
-                    model=str(model_path),
+            model_xml_path = self._get_model_xml_path(project_id, active_model_id)
+            model_bin_path = self._get_model_bin_path(project_id, active_model_id)
+            if not os.path.isfile(model_xml_path):
```

Comment:
I suggest 2 improvements to make it a bit more compact:
- make getter parametrized with type of model artifact (schema vs weights)
- move path verification there, so that it returns either valid response or raises exception

Reply:
Thanks for the suggestion, updated.